coding-test/wonji/level-1/체육복-2.js

```diff
@@ -0,0 +1,37 @@
+function solution(n, lost, reserve) {
+    let haveNum = n - lost.length; 
+  	// 전체 인원에서 도난당한 사람을 빼준다.
+  	// haveNum은 체육복을 가진 사람의 숫자이다.
+
+  	// 여벌의 체육복을 지녔지만 도난당한 사람이 있다면 
+  	// 배열에서 지우고 haveNum 카운트를 올린다.
+    for (let i = 0; i < lost.length; i++) {
+        if (reserve.indexOf(lost[i]) !== -1) {
+            reserve.splice(reserve.indexOf(lost[i]), 1);
+            lost.splice(i, 1);
+            haveNum ++;
+            i--;				
+          	// 중복되는 모든 경우를 제거하기 위해서 i값을 다시 -1 해주었다.
+        }
+    }
+	
+  	// 잃어버린 사람 앞이나 뒤에 여벌의 체육복을 가진 사람이 있다면,
+  	// 체육복을 빌리고, haveNum을 카운트한다.
+    for (let num of lost) {
+        let frontCheck = reserve.indexOf(num - 1)
+        let behindCheck = reserve.indexOf(num + 1)
+		
+    // 잃어버린 학생이 3, 5 / 여벌가진 학생이 4 였을때
+    // 4번 학생이 3번에게 빌려주고 
+    // 또 다시 5번에게 빌려주는 경우를 제거하기 위해서
+    // 빌려준 학생은 배열에서 제외시킨다.
+        if (frontCheck !== -1) {
+            reserve.splice(frontCheck, 1);
+            haveNum++;
+        } else if (behindCheck !== -1) {
+            reserve.splice(behindCheck, 1);
+            haveNum++;
+        }
```

Comment:
boolean 을 담은 변수값은 앞에 "is"를 붙여 표현하는 것이 컨벤션이다.
현업의 모든 개발자가 그렇게 하고있으니 주의하도록 하자. (현업에서 이렇게 안하면 그 회사 다니면 안됨)
isFront가 좋은 변수명으로 보이진 않기 때문에, Front보다 명확한 이름을 덧붙혀주는 것이 좋을 듯 하다. (코테 때는 상관없음)
추가로, DB에서는 boolean을 표현하는 변수를 front_flag와 같이 뒤에 "flag"를 붙인다. (이는 회사마다 다를 수 있으나 통상적으로 이런 추세를 갖고있다.)

위와 마찬가지로 배열의 원소 포함 여부는 indexOf가 아닌 includes를 사용하도록 하자.
카운팅 변수인 haveNum도 좋은 변수명은 아니다.
numCount와 같은 이름을 사용하는 것이 바람직하겠다.
```suggestion
    for (let num of lost) {
        let isFront = reserve.includes(num - 1)
        let isBehind = reserve.includes(num + 1)
		
    // 잃어버린 학생이 3, 5 / 여벌가진 학생이 4 였을때
    // 4번 학생이 3번에게 빌려주고 
    // 또 다시 5번에게 빌려주는 경우를 제거하기 위해서
    // 빌려준 학생은 배열에서 제외시킨다.
        if (isFront) {
            reserve.splice(isFront, 1);
            haveNum++;
        } else if (isBehind) {
            reserve.splice(isBehind, 1);
            haveNum++;
        }
```

Reply:
넵!